.github/workflows/build-and-run-tests.yml

```diff
@@ -0,0 +1,45 @@
+# This workflow will do a clean installation of node dependencies, cache/restore them, build the source code and run tests across different versions of node.
+# This workflow will be run whenever a new commit is pushed or when a PR is opened up.
+# Guide: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: Build and Run Tests
+
+on:
+  # run this for every push (CI)
+  push:
+  # also run this for PR
+  pull_request:
```

Comment:
Using both the `push` and `pull_request` made this workflow run twice when I opened up this PR...
Not sure what's the best approach here...
The intent was to make sure our own commits, as well PRs be tested.